roomserver/storage/postgres/state_snapshot_table.go

```diff
@@ -72,9 +72,31 @@ const bulkSelectStateBlockNIDsSQL = "" +
 	"SELECT state_snapshot_nid, state_block_nids FROM roomserver_state_snapshots" +
 	" WHERE state_snapshot_nid = ANY($1) ORDER BY state_snapshot_nid ASC"
 
+// Looks up both the history visibility event and relevant membership events from
+// a given domain name from a given state snapshot. This is used to optimise the
+// helpers.CheckServerAllowedToSeeEvent function.
+// TODO: There's a sequence scan here because of the hash join strategy, which is
+// probably O(n) on state key entries, so there must be a way to avoid that somehow.
+const bulkSelectStateForHistoryVisibilitySQL = "" +
```

Comment:
I would just use \` personally so you don't have to use `"` and `+` everywhere, makes it a bit easier to read at the cost of a bit of whitespace.